content/updates/terraform-null-label.md

```diff
@@ -0,0 +1,194 @@
+
+## The Importance of Consistent Naming
+
+In the early stages of a startup (or any project stack), the infrastructure is often simple and straightforward. The server and database are probably named something like `server01` and `database01`, respectively. The configuration files are easily accessible, and the team, being small, is intimately familiar with the system's architecture. The naming conventions, at this point, are uncomplicated and intuitive.
+
+However, as the project grows and evolves, so does its infrastructure. The once simple system expands to accommodate multiple customer-facing applications and a dozen other resources enter the mix. Infrastructure as Code (IaC) is implemented to manage and provision the increasingly complex tech stack. Amidst this growth, `database01` persists, but what about `database10`?. It's now `database10-us-east1-prod`; absolutely more descriptive and useful, but no one wants to rename `database01 `because it’s probably hardcoded into more than a few places.
+
+The issue of inconsistent naming becomes even more pronounced when we consider tags. What was once simply "backend" now has new variations like "backend-auth" and "backend-job-processor". Are those cloud resources getting tagged correctly so whoever is paying the bill knows that the "job processor" is costing 10x what the "auth" service is? Probably not.
+
+As the stack expands and you need to hire more engineers, the cognitive load and spool-up time for each new hire grows and grows. As it becomes harder to understand and categorize and "know about" the infrastructure, it becomes a bigger security/attack surface. Eventually billing and resource attribution becomes a problem too.
+
+## The Fix: terraform-null-label
+
+Well, the **real** fix is consistent naming and tagging, but the challenge has always been getting a growing, changing roster of engineers to decide on, implement, and enforce a standard _over time_.
+
+Good luck with that. Maybe you’ve written down your standards in documentation, but that’s hard to find, rarely read, and not easily enforceable.
+
+Why not use code?
+
+The `terraform-null-label` module, available at [https://github.com/cloudposse/terraform-null-label](https://github.com/cloudposse/terraform-null-label), offers a much better solution. We’ll look at some example usage in the next section but at a high level It allows users to define up to six primary inputs, referred to as "ID elements". These include:
+
+1. **Namespace:** A short abbreviation of the company name, typically 2-4 letters.
+2. **Tenant:** This is used for resources dedicated to a specific customer or team. (For a team like "data analytics" => "da")
+3. **Environment:** This refers to the AWS region. (Note that this nomenclature may be inverse to what most engineers are used to).
```

Comment:
```suggestion
3. **Environment:** This refers to the AWS region (note that this nomenclature may be inverse to what most engineers are used to).
```

Reply:
Just for a bit of history, no need to add it. 

Technically, this is how Cloud Posse has been using `environment` for our purposes. We struggled on what to use or adding another parameter like `region`. We didn't want to use region because the canonical region (e.g. `us-east-1`) is frequently needed in modules but is not the "region" used for naming/labeling purposes due to length constraints.

I don't see any reason why others cannot use `environment` to mean what we call stage. 

Of course, in `atmos` some of these are more opinionated, but that's something we intend to eliminate in Atmos soon, so arbitrary contexts can be used.